ci/scripts/cpp_build.sh

```diff
@@ -174,6 +174,11 @@ elif [ "${ARROW_EMSCRIPTEN:-OFF}" = "ON" ]; then
     -DCMAKE_UNITY_BUILD=${CMAKE_UNITY_BUILD:-OFF} \
     ${ARROW_CMAKE_ARGS} \
     ${source_dir}
+elif [ -n "${CMAKE_PRESET}" ]; then
+  cmake \
+    --preset="${CMAKE_PRESET}" \
+    ${ARROW_CMAKE_ARGS} \
+    ${source_dir}
```

Comment:
We want to provide preset for JNI build to share build options in apache/arrow and apache/arrow-java.